snapcraft/internal/db/migration.py

```diff
@@ -0,0 +1,116 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import abc
+import logging
+from datetime import datetime
+from typing import Any, ClassVar, Dict
+
+import tinydb
+
+import snapcraft
+
+logger = logging.getLogger()
+
+
+def get_snapcraft_version() -> str:
```

Comment:
why not pass snapcraft_version=snapcraft.__version__ in Migration.__init__, mocking becomes less of a pain and the coupling would be a bit more obvious.

Reply:
Sure

Reply:
did the same with Datastore, leaving the default there and no-default for Migration